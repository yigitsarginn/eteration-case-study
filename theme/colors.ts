export const colors = {
  black: '#000000',
  white: '#FFFFFF',
  blue: '#2A59FE',
  yellow: '#FFB800',
  lightGrey: '#D9D9D9',
  grey: '##C4C4C4',
  red: '#F90000',
  transparent: 'transparent',
};

export const rgba = (color: Colors, alpha: number): string => {
  const themeColor = colors[color];
  const hex = themeColor.replace('#', '');
  const r = parseInt(hex.substring(0, 2), 16);
  const g = parseInt(hex.substring(2, 4), 16);
  const b = parseInt(hex.substring(4, 6), 16);
  return `rgba(${r},${g},${b},${alpha})`;
};

export type Colors = keyof typeof colors;
