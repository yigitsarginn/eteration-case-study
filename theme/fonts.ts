import type { TextStyle } from 'react-native';

import { moderateScale } from '@/utils/layout-utils';

export type Fonts =
  | 'tiny'
  | 'xSmall'
  | 'small'
  | 'regular'
  | 'medium'
  | 'regularPlus';

export type FontWeightOptions =
  | 'light' // 300
  | 'regular' // 400
  | 'medium' // 500
  | 'semibold' // 600
  | 'bold' // 700
  | 'extraBold' // 800
  | 'black'; // 900

export type FontWeightValue = TextStyle['fontWeight'];

export type FontProps = {
  [key in Fonts]: {
    fontSize: number;
    fontWeight: FontWeightValue;
    lineHeight: number;
  };
};

export type FontWeightStyles = {
  [key in FontWeightOptions]: FontWeightValue;
};

export const fontFamily = {
  regular: 'Montserrat',
};

export const fonts: FontProps = {
  tiny: {
    fontWeight: '400',
    fontSize: moderateScale(12),
    lineHeight: moderateScale(14),
  },
  xSmall: {
    fontWeight: '400',
    fontSize: moderateScale(14),
    lineHeight: moderateScale(17.07),
  },
  small: {
    fontWeight: '400',
    fontSize: moderateScale(16),
    lineHeight: moderateScale(19.5),
  },
  regular: {
    fontWeight: '400',
    fontSize: moderateScale(18),
    lineHeight: moderateScale(21.94),
  },
  regularPlus: {
    fontWeight: '400',
    fontSize: moderateScale(20),
    lineHeight: moderateScale(24),
  },
  medium: {
    fontWeight: '400',
    fontSize: moderateScale(24),
    lineHeight: moderateScale(29.26),
  },
};

export const fontWeights: FontWeightStyles = {
  light: '300',
  regular: '400',
  medium: '500',
  semibold: '600',
  bold: '700',
  extraBold: '800',
  black: '900',
};
