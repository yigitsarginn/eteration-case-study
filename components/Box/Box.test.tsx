import React from 'react';
import { render } from '@testing-library/react-native';
import { Box } from './Box';
import { colors } from '@/theme';

describe('Box Component', () => {
  it('renders correctly with default props', () => {
    const { toJSON } = render(<Box testID="box" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('applies flex and padding props correctly', () => {
    const { getByTestId } = render(<Box testID="box" flex={1} p={10} />);
    const box = getByTestId('box');
    expect(box.props.style).toMatchObject({
      flex: 1,
      padding: 10,
    });
  });

  it('applies row style when row prop is true', () => {
    const { getByTestId } = render(<Box testID="box" row />);
    const box = getByTestId('box');
    expect(box.props.style).toMatchObject({
      flexDirection: 'row',
    });
  });

  it('applies center alignment when center prop is true', () => {
    const { getByTestId } = render(<Box testID="box" center />);
    const box = getByTestId('box');
    expect(box.props.style).toMatchObject({
      justifyContent: 'center',
      alignItems: 'center',
    });
  });

  it('applies background color correctly', () => {
    const { getByTestId } = render(<Box testID="box" color="transparent" />);
    const box = getByTestId('box');
    expect(box.props.style).toMatchObject({
      backgroundColor: colors['transparent'],
    });
  });

  it('combines multiple styles correctly', () => {
    const { getByTestId } = render(
      <Box
        testID="box"
        row
        center
        flex={1}
        p={10}
        m={5}
        radius={8}
        color="transparent"
      />,
    );
    const box = getByTestId('box');
    expect(box.props.style).toMatchObject({
      flex: 1,
      padding: 10,
      margin: 5,
      borderRadius: 8,
      backgroundColor: colors['transparent'],
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
    });
  });
});
