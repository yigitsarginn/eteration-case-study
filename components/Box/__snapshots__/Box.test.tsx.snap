// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Component renders correctly with default props 1`] = `
<View
  style={
    {
      "alignItems": undefined,
      "alignSelf": undefined,
      "backgroundColor": "transparent",
      "borderRadius": undefined,
      "flex": undefined,
      "gap": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginHorizontal": undefined,
      "marginVertical": undefined,
      "padding": undefined,
      "paddingHorizontal": undefined,
      "paddingVertical": undefined,
      "width": undefined,
    }
  }
  testID="box"
/>
`;
